api/envoy/config/accesslog/v3/accesslog.proto

```diff
@@ -282,6 +287,21 @@ message GrpcStatusFilter {
   bool exclude = 2;
 }
 
+// Filters based on matching dynamic metadata.
+// If the matcher path and key correspond to an existing key in dynamic metadata, the request is logged only if the matcher value is equal to the metadata value.
+// If the matcher path and key *do not* correspond to an existing key in dynamic metadata, the request is logged only if no_key_default is "true" or unset.
+message MetadataFilter {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.config.filter.accesslog.v2.MetadataFilter";
+
+  // Matcher to check metadata for specified value. For example, to match on the access_log_hint metadata,
+  // set the filter to "envoy.common" and the path to "access_log_hint", and the value to "true".
+  type.matcher.v3.MetadataMatcher matcher = 1;
+
+  // Default result if the key does not exist in dynamic metadata: if unset or true, then log; if false, then don't log.
+  google.protobuf.BoolValue no_key_default = 2;
```

Comment:
Sorry I think I understand why I was confused here. This is a filter, so this is about whether to log if the filter matches or not. I thought it was about a default value to log. My bad. Given that, I'm not sure whether true or false is the right default. I'm fine either way. But I think we should call this `match_if_key_not_found` or something like that? I think that would be more clear? wdyt?

Reply:
Yeah, I think that would be more clear actually. I'll change it to that.